hoodie-common/src/main/java/com/uber/hoodie/common/table/log/HoodieCompactedLogRecordScanner.java

```diff
@@ -63,69 +68,100 @@
   private AtomicLong totalLogRecords = new AtomicLong(0);
   // Total final list of compacted/merged records
   private long totalRecordsToUpdate;
+  // Latest valid instant time
+  private String latestInstantTime;
 
   public HoodieCompactedLogRecordScanner(FileSystem fs, List<String> logFilePaths,
-      Schema readerSchema) {
+      Schema readerSchema, String latestInstantTime) {
     this.readerSchema = readerSchema;
+    this.latestInstantTime = latestInstantTime;
 
+    // Store only the last log blocks (needed to implement rollback)
+    Deque<HoodieLogBlock> lastBlocks = new ArrayDeque<>();
```

Comment:
Is there a case where this queue will have more than 1 block? If not, can this be a variable?

Reply:
I had this as a variable earlier but there's a corner case which will create a problem : 
1. ingestionBatch() -> consists of updates + deletes for a fileId.
2. ingestionBatch() writes 2 blocks : 1 data block and 1 delete block.
3. Write fails, rollback()
4. rollback() -> writes 2 command blocks 1 for data and 1 for delete.
5. `HoodieCompactedLogRecordScanner` -> Needs to have more than 1 block to rollback() here.
Essentially, if there is a contract that each ingestionBatch() writes only one logBlock, we can do with a variable but for deletes it's not true. Another thought is to handle deletes differently ?

Reply:
Good points.. Lets have this as it is and please move the above as a comment, so future readers can follow along..

Reply:
I'll add this to the doc/issue where we explained the approaches.

Reply:
okay understand logging two rollback blocks now